### DISCUSSION TITLE
how to change kitQLClient URL during runtime with process.env?

### DISCUSSION THREAD

**Discussion by N3rdArtiste** [jycouet/kitql#155] [Q&A] [answered]
Session store is not accessible in non .svelte files. so I am unable to access it in kitQLClient.ts file.

**Comment by jycouet** [comment]
Hello,
I'm not sure to understand the link between the title and the description?

You want to change url via env file? You have multiple graphql endpoint? Or you want to set it once via env?

**Reply by N3rdArtiste** [reply]
I want to set kitQLClient URL through node variable while running my built sveltekit app.
like:
`GRAPHQL="https://example.com/graphql" node index.js`

**Reply by jycouet** [reply]
Not sure, but hopefully it helps 👇

for security reasons, vite don't provide env variables to your front-end by default. You have to 'opti-in' for that by prefixing env by VITE_ to make it available in your front-end.
Then, use it with: `url: import.meta.env.VITE_GRAPHQL`

Let me know if it helps

**Comment by jycouet** [comment, marked as answer]
Check out the [Announcement here](https://github.com/jycouet/kitql/discussions/156#discussioncomment-3080343)

---

Now you can hook up before or after the fetch, 

Here is an example 👇
```ts
import type { RequestHandlerArgs } from '$houdini';
import { HoudiniClient } from '$houdini';
import { stry } from '@kitql/helper';

// For Query & Mutation
async function fetchQuery({
  fetch,
  text = '',
  variables = {},
  session,
  metadata
}: RequestHandlerArgs) {
  // Prepare the request
  const url = import.meta.env.VITE_GRAPHQL_ENDPOINT || 'http://localhost:4000/graphql';

  // regular fetch (Server & Client)
  const result = await fetch(url, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      Authorization: `Bearer ${session?.token}` // session usage example
    },
    body: JSON.stringify({
      query: text,
      variables
    })
  });

  // return the result as a JSON object to Houdini
  const json = await result.json();

  // metadata usage example
  if (metadata?.logResult === true) {
    console.info(stry(json, 0));
  }

  return json;
}

// Export the Houdini client
export const houdiniClient = new HoudiniClient(fetchQuery);

```
